stats/cloud/api_test.go

```diff
@@ -202,3 +218,31 @@ func TestRetrySuccessOnSecond(t *testing.T) {
 	assert.NotNil(t, resp)
 	assert.Nil(t, err)
 }
+
+func TestSessionToken(t *testing.T) {
+	sessionToken := ""
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == http.MethodPost {
+			gotK6SessionToken := r.Header.Get(k6SessionTokenHeader)
+			if sessionToken == "" {
+				sessionToken = gotK6SessionToken
+			} else {
+				assert.NotEmpty(t, gotK6SessionToken)
+				assert.Equal(t, sessionToken, gotK6SessionToken)
+			}
+		}
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer server.Close()
+
+	client := NewClient("token", server.URL, "1.0")
+	client.retryInterval = 1 * time.Millisecond
+	req, err := client.NewRequest(http.MethodPost, server.URL, nil)
+	assert.NoError(t, err)
+	req.Header.Set(k6SessionTokenHeader, "xxx")
+	assert.NoError(t, client.Do(req, nil))
+
+	req, err = client.NewRequest(http.MethodGet, server.URL, nil)
+	assert.NoError(t, err)
+	assert.NoError(t, client.Do(req, nil))
+}
```

Comment:
Not certain what this test adds that the others you have changed don't already do ...

Reply:
@MStoykov this test for 2 case:

 - POST request with user supplies session token success.
 - Non-POST request success without session token.

Reply:
1. it doesn't test that the token that is set manually is the one that is used
2. it doesn't test that the Non-POST request don't send a token
So in both cases those *are* tested by other tests already

Reply:
@MStoykov Fair point, but I think it just does mean that the test does not do the actual meaningful test, not mean `both cases those are tested by other tests already`

I'm sure because you can check code cov log, I add the test above because code cov complains about missing of 2 above cases.

Anyway, updated PR coming.